shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/statement/ReadwriteSplittingPreparedStatementTest.java

```diff
@@ -92,4 +94,31 @@ public void assertGetGeneratedKeysWithPrimaryKeyIsNull() throws SQLException {
             assertFalse(generatedKeys.next());
         }
     }
+
+    @Test
+    public void assertGetGeneratedKeysWithPrimaryKeyIsNullInTransactional() throws SQLException {
+        try (
+                Connection connection = getReadwriteSplittingDataSource()
+                        .getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement("INSERT INTO t_config(id, status) VALUES(?, ?);", Statement.RETURN_GENERATED_KEYS)) {
+            connection.setAutoCommit(false);
+            Object lastIdObject = null;
+            Object idObject = null;
+            for (int i = 1; i <= 3; i++) {
+                preparedStatement.setObject(1, null);
+                preparedStatement.setString(2, "OK");
+                preparedStatement.executeUpdate();
+                ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+                assertTrue(generatedKeys.next());
+                int columnCount = generatedKeys.getMetaData().getColumnCount();
+                for (int index = 0; index < columnCount; index++) {
```

Comment:
Can you remove this loop logic? Just call `generatedKeys.getObject(1)` once.

Reply:
> Can you remove this loop logic? Just call `generatedKeys.getObject(1)` once.

I saw the previous test case and imitated it.I have changed them now.